buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt

```diff
@@ -77,230 +56,120 @@ limitations under the License."""
 open class IdePlugin : Plugin<Project> {
 
     override fun apply(project: Project): Unit = project.run {
-        configureIdeaForRootProject()
+        if (providers.environmentVariable("idea.active").getOrElse("false").toBoolean()) {
+            configureIdeaForRootProject()
+        }
     }
 
     private
     fun Project.configureIdeaForRootProject() {
+        val rootProject = this
         apply(plugin = "org.jetbrains.gradle.plugin.idea-ext")
-        apply(plugin = "idea")
-        tasks.named("idea") {
-            doFirst {
-                throw RuntimeException("To import in IntelliJ, please follow the instructions here: https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md#intellij")
```

Comment:
We don't need this anymore, since the `idea` task isn't added? Should we add a placeholder task which throws the exception?

Reply:
Added this back.